src/main/scala/scalismo/numerics/PivotedCholesky.scala

```diff
@@ -183,12 +183,19 @@ object PivotedCholesky {
     computeApproximateCholeskyGeneric[Int](kernel, indices, stoppingCriterion)
   }
 
-  private def computeApproximateEigGeneric[A](k: (A, A) => Double, xs: IndexedSeq[A], D: Double, sc: StoppingCriterion) = {
+  private def computeApproximateEigGeneric[A](k: (A, A) => Double, xs: IndexedSeq[A], sc: StoppingCriterion) = {
 
     val PivotedCholesky(l, _, _) = computeApproximateCholeskyGeneric(k, xs, sc)
 
-    val LD = l(::, 0 until l.cols).t *:* D
-    val phi: DenseMatrix[Double] = LD * l(::, 0 until l.cols)
+    // Parameter D is called mass matrix in the 'Galerkin discretization' of the eigenvalue problem.
+    // Here it can be written as a scalar value.
+    // For more information in regards to situations where D might be adapted, please start with
+    // "On the Low-rank Approximation by the PivotedCholesky Decomposition" (H. Harbrecht, M. Peters, R. Schneider)
+
```

Comment:
I find this comment somewhat confusing. Why is the matrix a scalar in this case? Where in the paper can we find the explanation? At least a first quick look at the paper does not reveal a discussion regarding the parameter D.